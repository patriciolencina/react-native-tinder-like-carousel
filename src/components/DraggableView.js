import React from 'react';
import {Animated, PanResponder} from 'react-native';

export default function Drag({children, moveLeft, moveRight, containerStyle}) {
  const pan = React.useRef(new Animated.ValueXY());

  const onMove = React.useCallback((_, {dx}) => {
    if (dx > 100) {
    } else if (dx < -100) {
    } else {
    }
  }, []);

  const panResponder = React.useMemo(
    () =>
      PanResponder.create({
        onStartShouldSetPanResponder: () => true,

        onPanResponderMove: Animated.event(
          [
            null,
            {
              dx: pan.current.x,
              dy: pan.current.y,
            },
          ],
          {
            listener: onMove,
          },
        ),
        onPanResponderRelease: (e, {dx}) => {
          if (dx > 100) {
            Animated.spring(pan.current, {toValue: {x: 500, y: 0}}).start(
              animated => {
                if (animated.finished) {
                  moveRight();
                }
              },
            );
          } else if (dx < -100) {
            Animated.spring(pan.current, {toValue: {x: -500, y: 0}}).start(
              animated => {
                if (animated.finished) {
                  moveLeft();
                }
              },
            );
          } else {
            Animated.spring(pan.current, {toValue: {x: 0, y: 0}}).start();
          }
        },
      }),
    [moveLeft, moveRight, onMove],
  );

  return (
    <Animated.View
      {...panResponder.panHandlers}
      style={[pan.current.getLayout(), containerStyle]}>
      {children}
    </Animated.View>
  );
}
