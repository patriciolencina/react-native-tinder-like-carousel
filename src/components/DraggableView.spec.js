import React from 'react';
import {shallow} from 'enzyme';
import DraggableView from './DraggableView';

describe('DraggableView', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      expect(shallow(<DraggableView />)).toMatchSnapshot();
    });
  });
});
