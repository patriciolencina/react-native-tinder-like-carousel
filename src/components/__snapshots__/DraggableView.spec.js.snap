// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraggableView Rendering should match to snapshot 1`] = `
<AnimatedComponent
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Array [
      Object {
        "left": 0,
        "top": 0,
      },
      undefined,
    ]
  }
/>
`;
